File: spec/test_lexer_multiple_modes.c
#include "testparser.h"
#include <assert.h>
#include <string.h>
#include <stdio.h>

int main()
{
    char const * input = "abc.def";
    p_context_t * context;
    context = p_context_new((uint8_t const *)input, strlen(input));
    assert(p_parse(context) == P_SUCCESS);
    printf("pass1\n");
    p_context_delete(context);

    input = "abc .  abc";
    context = p_context_new((uint8_t const *)input, strlen(input));
    assert(p_parse(context) == P_SUCCESS);
    printf("pass2\n");
    p_context_delete(context);

    return 0;
}

File: spec/test_lexer_multiple_modes.rs
use testparser::*;

fn main() {
    let mut context = p_context_new(b"abc.def");
    assert_eq!(P_SUCCESS, p_parse(&mut context));
    println!("pass1");
    p_context_delete(context);

    let mut context = p_context_new(b"abc .  abc");
    assert_eq!(P_SUCCESS, p_parse(&mut context));
    println!("pass2");
    p_context_delete(context);
}
